query: new function for showing running config

cathay4t opened the pull request:
Introduced `libnmstate.show_running_config()` for querying
activated/running network configuration excluding:
* IP address retrieved by DHCP or IPv6 auto configuration.
* DNS client resolver retrieved by DHCP or IPv6 auto configuration.
* Routes retrieved by DHCPv4 or IPv6 router advertisement.
* LLDP neighbor information.

The output will use the same schema of `libnmstate.show().

The `-r` and `--running-config` option has been added to `show` subcommand
of nmstatectl.

The `io.nmstate.ShowRunningConfig` function has been added to varlink
interface of nmstatectl.

Integration test cases are included for API, CLI and varlink.

cathay4t commented:
~~Wait on PR #1418~~

sakalosj commented:
/packit build

ffmancera reviewed (changes requested):
Thanks, this looks very good. Could you add the option to the manpage? Thanks!

cathay4t commented:
Manpage updated. Thanks.

ffmancera reviewed (approved):
LGTM, thanks!

cathay4t committed:
query: new function for showing running config

Introduced `libnmstate.show_running_config()` for querying
activated/running network configuration excluding:
 * IP address retrieved by DHCP or IPv6 auto configuration.
 * DNS client resolver retrieved by DHCP or IPv6 auto configuration.
 * Routes retrieved by DHCPv4 or IPv6 router advertisement.
 * LLDP neighbor information.

The output will use the same schema of `libnmstate.show().

The `-r` and `--running-config` option has been added to `show`
subcommand of nmstatectl.

The `io.nmstate.ShowRunningConfig` function has been added to varlink
interface of nmstatectl.

Manpage of `nmstatectl` updated.
Integration test cases are included for API, CLI and varlink.

Signed-off-by: Gris Ge <fge@redhat.com>
